lighthouse-cli/test/smokehouse/report-assert.js

```diff
@@ -285,6 +308,7 @@ function assertLogString(count) {
  */
 function report(actual, expected, reportOptions = {}) {
   const localConsole = new LocalConsole();
+  setCurrentLhr(actual.lhr);
```

Comment:
filtering expectations makes more sense to me, threading also seems fine to me.

this route is my least favorite of the 3 sorry

Reply:
> filtering expectations makes more sense to me, threading also seems fine to me.

+1, though threading also sounds like a lot work, so I'd prefer filtering as well.

Simpler is better, and we don't need a ton of functionality here. Quick fourth option: rather than have any possible expectations level filterable, what if we
- had `minChromeVersion` be on the top level `Smokehouse.TestDefn`
- did a test matrix with stable and canary (which we should probably be doing anyways rather than relying on Travis/GH/Appveyor) and skipped tests where `minChromeVersion` didn't match

Reply:
> had minChromeVersion be on the top level Smokehouse.TestDefn

I don't want to force another run just for this (see the smoke test in perf that was enabled). and it would require more than just deleting a single line to reverse.

> did a test matrix with stable and canary (which we should probably be doing anyways rather than relying on Travis/GH/Appveyor) and skipped tests where minChromeVersion didn't match

Would need to install stable and canary in each CI env, right? Can we? This seems good as an additional enhancement, but I don't see how this is orthogonal to the first the options.

Reply:
> I don't want to force another run just for this (see the smoke test in perf that was enabled)

Github actions should be able to run them in parallel (I assume we would be soaking up our concurrency quota, but we seem to have overhead still)

> it would require more than just deleting a single line to reverse.

Once Chrome 85 or whatever hits stable, what else would we need to do but delete the one `minChromeVersion` line? Or am I missing what you're saying?

> Would need to install stable and canary in each CI env, right? Can we? This seems good as an additional enhancement, but I don't see how this is orthogonal to the first the options.

We can in GH and Travis, for sure. We probably wouldn't want to bother in AppVeyor. But in my suggestion it wouldn't be orthogonal, it would be what lets it run all the smoke tests (those with a min version in Canary and those without in Stable and Canary).

Reply:
> Once Chrome 85 or whatever hits stable, what else would we need to do but delete the one minChromeVersion line? Or am I missing what you're saying?

yeah, look at the perf expectation I commented out. it's not the only thing being tested on the URL, but it is the only thing that requires M84. Moving the minimum version declaration to only toplevel would require splitting that expectation in two.

> But in my suggestion it wouldn't be orthogonal, it would be what lets it run all the smoke tests (those with a min version in Canary and those without in Stable and Canary).

I think we're on the same page, just bogged down on wording. Running these tests for both canary and stable is def. something we should do.

It'd require being able to skip tests based on the Chrome version, and either options 1-3 would work for that purpose. I can take a crack at this after this PR lands (just implemented the filtering option).